### PR TITLE
Csv download

### LINKED ISSUE #30: Export results as CSV
Via Catherine Marshchner.

### PR DESCRIPTION
**Pull request by Lathrisk**

### CODE REVIEW

File: DataQualityTester/views/quality.py [hunk 1]
```diff
@@ -143,19 +143,19 @@ def download_results(task_id):
...
         with open(component_file) as f:
             results = json.load(f)
             category_score = _compute_score(results)
-        csv_file.writerow(['category', component.name, category_score])
+        csv_file.writerow(['category', '', component.name, category_score])
 
         for test, test_scores in results.items():
             test_score = percent(test_scores)
-            print(test)
+            indicator = INDICATOR_LOOKUP[test]
```
Comment (michaelwood) [inline DataQualityTester/views/quality.py]: Getting a key error on some results for this dict.

I used a try/catch block as we know this lookup can fail. 
https://github.com/pwyf/data-quality-tester/blob/2021-develop/DataQualityTester/views/quality.py#L67

Might be worth abstracting that into a "get_indicator_num" and putting the catch in there.. 
Perhaps there is also one title missing from the spreadsheet/different in the spreadsheet from the test.

```
Traceback (most recent call last):
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/DataQualityTester/routes.py", line 52, in download_results
    response_csv = quality.download_results(uuid)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/DataQualityTester/views/quality.py", line 156, in download_results
    indicator = INDICATOR_LOOKUP[test]
KeyError: 'Country strategy or Memorandum of Understanding'
```

Comment (Lathrisk) [reply DataQualityTester/views/quality.py]: Thanks, I didn't get that error, I wonder if the indicator definitions are different. I've added a try/catch block as per your previous changes.

File: DataQualityTester/templates/partials/_test_another_file.html [hunk 2]
```diff
@@ -2,5 +2,8 @@
   <div class="col-lg-2">
     <a href="{{ url_for('home') }}" class="btn btn-default">Test another file</a>
   </div>
+  <div class="col-lg-2">
+    <a href="{{ url_for('download_results', uuid=uuid) }}" class="btn btn-default">Download results</a>
+  </div>
   <div class="col-lg-offset-5 col-lg-5"></div>
 </div>
```
Comment (michaelwood) [inline DataQualityTester/templates/partials/_test_another_file.html]: A nice to have would be for this column layout to be fixed. The offset-5  div is a bit weird, you'd also think that would need decrementing, but also there are other strange things going on with padding/margins to stop the buttons from overlapping and that is why they sit so far apart. Kind of odd, unfortunately forgotten most of my bootstrap 3 knowledge.

File: DataQualityTester/views/quality.py [hunk 3]
```diff
@@ -132,3 +133,30 @@ def task_status(task_id):
...
+
+        for test, test_scores in results.items():
+            test_score = percent(test_scores)
+            csv_file.writerow(['test', test, test_score])
+
+        csv_file.writerow(['','',''])
+
+    output = make_response(csv_response.getvalue())
+
+    return output
```
Comment (michaelwood) [inline DataQualityTester/views/quality.py:170]: Haven't been able to test this yet

**Review by michaelwood — CHANGES_REQUESTED** [review]
Key error main issue

**Review by michaelwood — APPROVED** [review]
lgtm - and have now tested download.

**Pull request merged**